src/overlays/actors/ovl_Bg_Keikoku_Saku/z_bg_keikoku_saku.c

```diff
@@ -28,21 +32,70 @@ const ActorInit Bg_Keikoku_Saku_InitVars = {
     (ActorFunc)BgKeikokuSaku_Draw,
 };
 
-#endif
+void BgKeikokuSaku_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgKeikokuSaku* this = THIS;
+    s32 pad;
+    CollisionHeader* colHeader;
+
+    colHeader = NULL;
+    DynaPolyActor_Init(&this->Dyna, 0);
+    CollisionHeader_GetVirtual(&object_keikoku_obj_Colheader_002300, &colHeader);
+    this->Dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->Dyna.actor, colHeader);
+    this->unk160 = BGKEIKOKUSAKU_GET_SWITCHFLAG(thisx);
+    if (Flags_GetSwitch(globalCtx, this->unk160)) {
+        this->Dyna.actor.world.pos.z = 2659.0f;
+        return;
+    }
+    this->actionFunc = func_80A5389C;
+}
 
-extern UNK_TYPE D_06001640;
-extern UNK_TYPE D_06002300;
+void BgKeikokuSaku_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgKeikokuSaku* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Keikoku_Saku/BgKeikokuSaku_Init.s")
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->Dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Keikoku_Saku/BgKeikokuSaku_Destroy.s")
+void func_80A5389C(BgKeikokuSaku* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, this->unk160)) {
+        this->actionFunc = func_80A538E0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Keikoku_Saku/func_80A5389C.s")
+void func_80A538E0(BgKeikokuSaku* this, GlobalContext* globalCtx) {
+    Actor_PlaySfxAtPos(&this->Dyna.actor, NA_SE_EV_METALGATE_OPEN - SFX_FLAG);
+    this->Dyna.actor.world.pos.z -= 2.0f + gGameInfo->data[0x968];
```

Comment:
This `gGameInfo->data` accesses are special and have special macros for them. You should be able to use the `tools/regconvert.py` script to convert them.